R/AcqOptimizer.R

```diff
@@ -15,23 +15,80 @@ AcqOptimizer = R6Class("AcqOptimizer",
     initialize = function(optimizer, terminator) {
       self$optimizer = assert_r6(optimizer, "Optimizer")
       self$terminator = assert_r6(terminator, "Terminator")
+      ps = ParamSet$new(list(
+        ParamLgl$new("fix_distance"),
+        ParamDbl$new("dist_threshold", lower = 0, upper = 1))
+      )
+
+      ps$values = list(fix_distance = FALSE, dist_threshold = 0)
+      ps$add_dep("dist_threshold", on = "fix_distance", cond = CondEqual$new(TRUE))
+      private$.param_set = ps
     },
 
     #' @description
     #' Optimize the acquisition function.
     #'
-    #' @param acq_function [AcqFunction].
-    #' @return `data.table` with 1 row per optimum and x as colums
-    optimize = function(acq_function) {
-      if (acq_function$codomain$length == 1) {
+    #' If the `fix_distance` parameter is set to `TRUE`, proposed points are
+    #' replaced by randomly sampled ones if their Gower distance with respect
+    #' to other proposed points or previously evaluated points falls below the
+    #' `dist_threshold` parameter.  Note that these checks are only performed a
+    #' single time (and not after a potential replacement of points), i.e., the
+    #' returned [data.table::data.table()] of optima must not necessarily pass
+    #' the checks but most likely will.
+    #'
+    #' @param acq_function [AcqFunction]\cr
+    #' Acquisition function to optimize.
+    #' @param archive [bbotk::Archive]\cr
+    #' Archive.
+    #'
+    #' @return [data.table::data.table()] with 1 row per optimum and x as columns.
+    optimize = function(acq_function, archive) {
+      if (acq_function$codomain$length == 1L) {
         instance = OptimInstanceSingleCrit$new(objective = acq_function, terminator = self$terminator)
       } else {
         if (!"multi-crit" %in% self$optimizer$properties) {
           stopf("Optimizer %s is not multi-crit compatible but %s is multi-crit.", self$self$optimizer$format(), acq_function$id)
         }
         instance = OptimInstanceMultiCrit$new(objective = acq_function, terminator = self$terminator)
       }
-      self$optimizer$optimize(instance)
+
+      xdt = tryCatch(self$optimizer$optimize(instance),
```

Comment:
Here we don't know if the optimizer crashed or the surrogates' prediction?

Reply:
I agree, however, the error message is logged and should be different for surrogate vs. acq optimization failure